src/components/ColorGenerator.astro

```diff
@@ -0,0 +1,599 @@
+---
+import { Tabs, TabItem, Code } from "@astrojs/starlight/components";
+
+let rgbCppCode = "rgb_color(128, 128, 128);";
+let rgbCsharpCode = "SplashKit.RGBColor(128, 128, 128);";
+let rgbPythonCode = "rgb_color(128, 128, 128)";
+
+let hsbCppCode = "hsb_color(180.00, 0.50, 0.50);";
+let hsbCsharpCode = "SplashKit.HSBColor(180.00, 0.50, 0.50);";
+let hsbPythonCode = "hsb_color(180.00, 0.50, 0.50)";
+---
+
+<style>
+  /* General slider styles */
+  input[type="range"] {
+    -webkit-appearance: none; /* Remove default appearance */
+    width: 100%; /* Full width */
+    height: 10px; /* Height of the slider */
+    background: #ddd; /* Default background color */
+    border-radius: 5px; /* Rounded edges */
+    outline: none; /* Remove focus outline */
+    transition: background-color 0.3s ease; /* Smooth background color change */
+  }
+
+  /* Individual styling for Red slider */
+  #red-slider {
+    background-color: #f8d7d7; /* Light red background */
+  }
+  #red-slider:hover {
+    background-color: red; /* Intense red on hover */
+  }
+  #red-slider::-webkit-slider-thumb {
```

Comment:
Hi @ShaunR1991 , this is a wonderful PR and addition to the documentation site. It gives users the ability to quickly select the splashkit colours from the color palette and generate colors from the generator. I've reviewed the guide and it is well written too.

I have one change to suggest here on this style for the `#red-slider::-webkit-slider-thumb`, it uses custom styles that aren't consistent with the other slider thumb styles. Also the other slider thumbs for blue and green, there are `background-color` properties being changed but not actually applied as its missing the `-webkit-appearance: none; /* Remove default appearance */` style to remove the default styles first.

Here the red slider thumb has custom styles and the other slider thumbs have default styles
<img width="663" alt="image" src="https://github.com/user-attachments/assets/a4d6121e-7951-4c33-9704-37f62461f149" />

Reply:
Great catch @ctrlvish ! I was able to use this information when finalising this PR :)